Woof/WoofTests/Model/Preferences/PreferencesHandlerGetUserRoleTests.swift

```diff
@@ -0,0 +1,44 @@
+import XCTest
+
+final class PreferencesHandlerGetUserRoleTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.preferencesStorage
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testExistsInAPI() {
+        PreferencesHandler.getUserRole()
+    }
+
+    func testReturnsNoneForEmptyStorage() {
+        XCTAssertEqual(PreferencesHandler.getUserRole(), .none)
+    }
+
+    func testReturnsSavedRole() {
+        // Given
+        let role = Role.owner
+        PreferencesHandler.set(userRole: role)
+
+        // When
+        let loadedRole = PreferencesHandler.getUserRole()
+
+        // Then
+        XCTAssertEqual(loadedRole, role)
+    }
+
+    func testSavesLastRoleWhenSetMethodCalledMultiplyTimes() {
+        // Given
+        let repeatedRoles = Array(repeating: Role.allCases, count: 4).flatMap { $0 }
```

Comment:
nit:
Using  6 roles (repeating 2 times, or even 1 iteration) will be enough to test the functionality for me, why do we need to create the array of 12 roles?